### PR TITLE
New JSON-specific methods and accompanying test

### PR DESCRIPTION
**Pull request by chartjes**
@stuartphilp review?

### CODE REVIEW

File: kinto/api-test/mockclient.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 
 stage_url = "https://kinto.stage.mozaws.net/v1/"
 
-
```
Comment (stuartphilp) [inline kinto/api-test/mockclient.py:7]: it's a silly pep8 standard, but nevertheless it is https://www.python.org/dev/peps/pep-0008/#blank-lines

File: kinto/api-test/mockclient.py [hunk 2]
```diff
@@ -40,6 +39,13 @@ def post_request(self, resource, data=None, status_code=None):
             r.raise_for_status()
         return json.loads(r.content.decode('utf-8'))
 
+    def post_json_request(self, resource, data=None, status_code=None):
```
Comment (stuartphilp) [inline kinto/api-test/mockclient.py:43]: does this need to be a separate request type? maybe we can use the json.dumps method as in http://docs.python-requests.org/en/latest/user/quickstart/#more-complicated-post-requests

File: kinto/api-test/mockclient.py (hunk 1 above)
Comment (chartjes) [inline kinto/api-test/mockclient.py:7]: Will fix.

On Tue, Dec 1, 2015, at 02:53 PM, stuartphilp wrote:

> In kinto/api-test/mockclient.py[1]:
> 
> > @@ -4,7 +4,6 @@
> 
>  stage_url = "https://kinto.stage.mozaws.net/v1/"
> - it's a silly pep8 standard, but nevertheless it is
>   https://www.python.org/dev/peps/pep-0008/#blank-lines
> 
> — Reply to this email directly or view it on GitHub[2].

Links:
1. https://github.com/mozilla-services/services-test/pull/31#discussion_r46329447
2. https://github.com/mozilla-services/services-test/pull/31/files#r46329447

File: kinto/api-test/mockclient.py (hunk 2 above)
Comment (chartjes) [inline kinto/api-test/mockclient.py:43]: The json.dumps() method breaks other tests. The alternative is for me to
just manually use the requests library instead of adding a new method.
Either way I think we need at least one test that submits Unicode
strings to make sure Kinto handles them correctly

On Tue, Dec 1, 2015, at 02:57 PM, stuartphilp wrote:

> In kinto/api-test/mockclient.py[1]:
> 
> > @@ -40,6 +39,13 @@ def post_request(self, resource, data=None,
> > status_code=None):
> >              r.raise_for_status() return json.loads(r.content.decode('utf-
> >                8'))
> -    def post_json_request(self, resource, data=None,
>    status_code=None): does this need to be a separate request type?
>    maybe we can use the json.dumps method as in
>    http://docs.python-requests.org/en/latest/user/quickstart/#more-complicated-post-requests
> 
> — Reply to this email directly or view it on GitHub[2].

Links:
1. https://github.com/mozilla-services/services-test/pull/31#discussion_r46330028
2. https://github.com/mozilla-services/services-test/pull/31/files#r46330028

Comment (stuartphilp) [inline kinto/api-test/mockclient.py:43]: agreed, definitely want to test unicode. okay, this is fine

**Comment by stuartphilp** [comment]
r+ with the line added back in

**Comment by chartjes** [comment]
@stuartphilp I ran it through a PEP8 linter, please take a look.

**Comment by stuartphilp** [comment]
ah yes 80 character lines :) merging
